src/VisualStudio/Core/Def/Options/VisualStudioUnifiedSettingsOptionPersister.cs

```diff
@@ -27,50 +27,60 @@ private void OnSettingChanged(SettingsUpdate update)
             this.RefreshIfTracked(key);
     }
 
-    public override bool TryFetch(OptionKey2 optionKey, string storageKey, out object? value)
+    private static void CheckStorageKey(string storageKey)
     {
-        if (!TryFetchWorker(optionKey, storageKey, typeof(string), out var innerValue) ||
-            innerValue is not string innerStringValue)
-        {
-            value = null;
-            return false;
-        }
-
-        return optionKey.Option.Definition.Serializer.TryParse(innerStringValue, out value);
+        Contract.ThrowIfFalse(storageKey.StartsWith("languages"), "Need to update SubscribeToChanges in constructor to listen to changes to this key");
     }
 
-    public override Task PersistAsync(OptionKey2 optionKey, string storageKey, object? value)
+    internal override Optional<object?> TryReadOptionValue(OptionKey2 optionKey, string storageKey, Type storageType, object? defaultValue)
     {
-        // In-memory representation was different than persisted representation (often a bool/enum), so
-        // serialize it as per the option's serializer.
-        //
-        // Note, we persist as a lowercase value, as that's what the setting manager does for these modern keys. On
-        // read, TryParse will handle lowercase enum values just fine due to it using `Enum.TryParse(str,
-        // ignoreCase: true, out result)`
-        var serialized = optionKey.Option.Definition.Serializer.Serialize(value).ToLowerInvariant();
-        return this.PersistWorkerAsync(storageKey, serialized);
-    }
+        CheckStorageKey(storageKey);
 
-#pragma warning disable CS8714 // The type cannot be used as type parameter in the generic type or method. Nullability of type argument doesn't match 'notnull' constraint.
-    protected override bool TryGetValue<T>(string storageKey, out T value)
-    {
-        Debug.Assert(typeof(T) == typeof(string));
-        var retrieval = this.SettingsManager.GetReader().GetValue<T>(
-            storageKey, SettingReadOptions.NoRequirements);
+        if (storageType == typeof(int))
```

Comment:
the unified-settings codepath now gets simpler as well.  it just handles ints (for tab-size/indent-size), and enums (for use-spaces, and indent-style).  

If we move more things here, we can expand this out as appropriate.